docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,219 @@
+    the `API details` section on product and collection pages in the Shopify admin. 
+    `API details` will only be displayed if you have a private app with Storefront API access. For more details see the 
+    [Getting Started Guide](https://help.shopify.com/api/storefront-api/getting-started#authentication) for the Storefront API.
+
+2.  The functions now take an additional optional `query` argument that specifies the fields to query on the resource.
+    ```js
+    const query = productNodeQuery(['title', ['images', imageConnectionQuery()]];
+
+    client.fetchProduct('Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUzMTMwMjc=', query).then((product) => {
+      console.log(product); // Product with only title and images
+    });
+    ```
+    Fields and queries can be nested like so:
+    ```js
+    // Product query with title and variants, where variants have price and title
+    const query = productNodeQuery(['title', ['variants', variantConnectionQuery(['price', 'title'])]]);
+    ```
+    If the `query` argument isn't supplied, a default list of arguments will be used.
+    See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md#Client.Queries) for more details.
+
+3.  `fetchCollection()` and `fetchAllCollections()` no longer return the products in the collections by default.
+    To fetch products on the collections, use `fetchCollectionWithProducts()` and `fetchAllCollectionsWithProducts()`,
+    or specify products in the query function parameter.
+    ```js
+    const collectionId = 'Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI1NzY5NzczMQ=='
+    
+    // Use the built-in function
+    client.fetchCollectionWithProducts(collectionId).then((collection) => {
+      console.log(collection); // Collection with all default fields and products with all default fields.
+    });
+
+    // or specify a custom query
+    const query = collectionNodeQuery(['title', ['products', productConnectionQuery(['title'])]]);
+
+    client.fetchCollection(collectionId, query).then((collection) => {
+      console.log(collection); // Collection with only title and products with only title
+    });
+    ```
+    Note that `fetchCollectionWithProducts()` and `fetchCollectionsWithProducts()` do not take a `query` argument.
+
+4.  `fetchQueryProducts()` and `fetchQueryCollections()` query different fields and take an optional `query` argument.
+    See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md) for more details.
+
+### Carts/Checkouts
+
+Carts are replaced with checkouts. Like the fetch functions, all checkout functions take an optional `query` argument that specifies fields to return on the checkout.
+
+`fetchRecentCart()`, `updateModel()`, and `clearLineItems()` have been removed. `checkoutUrl` has been renamed to `webUrl`.
+
+#### Creating a Checkout
+
+To create a checkout, use `createCheckout()`.
```

Comment:
You should make a note somewhere either here, or around the removed functions, or in both places, that the SDK user is responsible for capturing the ID for later usage. If they want to persist the checkout between sessions, they should store the ID in a cookie or localStorage.

Since this is a migration guide, we should show them how to emulate past behaviour.